### ISSUE TITLE
Add CI check for vtadmin-web proto/typegen drift

### ISSUE THREAD

**Issue by doeg** [vitessio/vitess#9001]
## tl;dr

Right now, it's up to humans to remember to run `make vtadmin_web_proto_types` after modifying [vtadmin.proto](https://github.com/vitessio/vitess/blob/69e3766/proto/vtadmin.proto) or any of its dependencies. For branches that modify at least one [proto/*.proto](https://github.com/vitessio/vitess/tree/main/proto) file, I propose adding a CI check for drift between the generated types on the PR branch vs. the main branch. 

## Longer version

The [`vtadmin_web_proto_types` makefile target](https://github.com/vitessio/vitess/blob/main/Makefile#L460-L464) generates TypeScript declarations + .js wrappers from [vtadmin.proto](https://github.com/vitessio/vitess/blob/69e3766/proto/vtadmin.proto), into [vtadmin-web/src/proto/](https://github.com/vitessio/vitess/tree/14558dd/web/vtadmin/src/proto).

Anytime vtadmin.proto or any of its .proto dependencies are modified, it's necessary to run `make proto` (as usual) and _then_ `make vtadmin_web_proto_types`. 

Forgetting this step can result in vtadmin-web runtime errors, since vtadmin-web will be using different generated types, compared to vtadmin-api and the rest of Vitess.

This, combined with a CI check to confirm the front-end build is green (https://github.com/vitessio/vitess/pull/8993) should avoid situations like https://github.com/vitessio/vitess/pull/8623 and https://github.com/vitessio/vitess/pull/8566. In both cases, I wasn't around to review and it (understandably!!!) wasn't clear to the PR authors whether or not the changes broke/otherwise affected vtadmin-web.

@ajm188 mentioned that a proto diff/drift check could be more widely useful to other makefile targets. I wanted to keep this ticket scoped fairly narrowly to vtadmin-web since it's less work for me 😈 but if it's applicable elsewhere, then that's great + I'd love to know about it! 

## A solution + an alternative

Two potential solutions, both of which will address the problem of drift between vtadmin.proto and the front-end codegen. ...Well, more like: one good solution and then a less-good alternative that I will mention for completeness. :) 

✨  **Proposed solution:** Add a "proto drift check" in CI that runs make proto and make vtadmin_web_proto_types, and compares the resulting TypeScript/JavaScript codegen with the file on the branch. The check will fail if there are any diffs between the generated-in-CI version and the branch version. If the check fails, the dev will (for now) need to run `make vtadmin_web_proto_types` locally and then push up the newly generated files. 

Pros:
- As @ajm188 mentioned in our conversation, we could (now or later) broaden this check (or add a new one) to run a similar diff/drift check against other generated (golang) code.
- Devs won't need node/npm/other front-end dependencies to run `make proto` _unless_ the change would cause drift and/or legitimately break the front-end build.

Cons: 
- Setting up a drift check is a nonzero amount of work. (But I'm happy to do it!)
- It might be tricky to make the check only run if vtadmin.proto or any of its dependencies are edited, since that would require knowing which subset of the protos are used by vtadmin. Not impossible, but walking a dependency tree seems like overkill when we could (more easily, at the expense of a wider net) run the check on changes to _any_ proto/*.proto file.
- Fixing the build will necessitate running `make vtadmin_web_proto_types` _somewhere_. (If people are really resistant to installing the front-end dependencies, we can figure that out, too.) Not a con, per se, but worth mentioning. 

☹️  **Rejected unpleasant alternative:** the vtadmin_web_proto_types target to the make proto target (or add a new target), so that it will always be run without having to remember to do so. 

Pros:
- Humans won't have to remember to run `make vtadmin_web_proto_types`

Cons:
- `make proto` will have a "dependency" on the vtadmin_web_proto_types target, and therefore will have a dependency on the correct node version being installed. I can understand why this would be a big con for backend devs. :)
- vtadmin_web_proto_types would be run on any .proto file change, even if it's not a dependency of vtadmin.proto.
- Probably more cons beyond these since overall this seems cumbersome and annoying!
